### DISCUSSION TITLE
TS0601 Expected entities & attributes

### DISCUSSION THREAD

**Discussion by SamMousa** [zigpy/zha-device-handlers#1200] [Q&A]
I've recently gotten a moeshouse TRV. I've added it to ZHA and it shows 3 entities:
- Thermostat
- On_off
- Power

As far as I can tell it is properly detected by a quirk:
```js
{
  "manufacturer": "_TZE200_cwnjrr72",
  "model": "TS0601",
  "class": "zhaquirks.tuya.ts0601_trv.MoesHY368_Type1"
}
```

When manually exploring the clusters I can get more detailed information like valve open % and a cursory reading of the source code here: https://github.com/zigpy/zha-device-handlers/blob/dev/zhaquirks/tuya/ts0601_trv.py seems to suggest that it should be mapped to an entity attribute.

So to get to my questions:

1. What entities and attributes should be shown for a fully supported TRV
2. Where is this documented
3. Is the missing valve percentage attribute a bug or a not yet implemented feature? (This essentially follows from 1 and it impacts what kind of issue if any I should make)

**Comment by MattWestb** [comment]
One user have making one updated quirk for "Moes" TRVs (more correct CN-Hysen) that shall having more GUI entries in ZHA https://github.com/zigpy/zha-device-handlers/issues/1178#issuecomment-980294492.

If you like trying it and giving hi feedback its being great.

Its very large changes for all TRV and for the moment its not possible getting the update merged in to ZHA then its braking most old functions that some user is still using.

**Reply by MattWestb** [reply]
If you like to knowing what we have finding for functions in tuya TRVs you can taking one look on https://github.com/zigpy/zigpy/discussions/653#discussioncomment-314395 but we have not getting feedback from all users if all functions is working OK in the devices.
Some is today very badly supported with not implanted or broken functions (like my Siterwells) but updates is in the pipe to getting so much we can from our devices and with more GUI elements in ZHA for the user.

**Comment by SamMousa** [comment]
Thanks for your answers, I'll have a look and see what I can find!

**Comment by SamMousa** [comment]
I have tried the quirk and while it shows a lot of extra entities I'm not sure they are actually working properly. I'm getting wrong data and changing any of the entities doesn't work.
